### PR TITLE
BaSD-MS-64: add Interviews entity form and list table 

### PR DESCRIPTION
**Pull request by veronabaudracco**
Client entity:
Added components to show all Interview entities on a table, modal and form to edit and create a new one.

Valid functionalities:

- [x] CreateCreate
- [x] Visualize
- [x] Remove
- [x] Edit

### CODE REVIEW

File: src/Components/Interviews/Form/form.module.css [hunk 1]
```diff
@@ -0,0 +1,44 @@
+.container {
+  flex: 1;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  align-items: center;
+}
+select, input{
```
Comment (AgustinCarthery) [inline src/Components/Interviews/Form/form.module.css]: avoid tag name selectors

File: src/Components/Interviews/Form/index.js [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+          name="postulantId"
+          type="text"
+          required
+          value={postulantIdValue}
+          onChange={onChangePostulantId}
+        >
+          {postulantsValue.map((postulant) => {
+            return (
+              <option value={postulant._id} key={postulant._id}>
+                {postulant.firstName + ' ' + postulant.lastName}
```
Comment (AgustinCarthery) [inline src/Components/Interviews/Form/index.js]: literal string would suit better

File: package.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "@testing-library/user-event": "^12.8.3",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
-    "react-scripts": "4.0.3",
+    "react-scripts": "^4.0.3",
```
Comment (estefrare) [inline package.json]: do not update this library

File: src/Components/Interviews/Form/index.js [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+          return response.json().then(({ message }) => {
+            throw new Error(message);
+          });
+        }
+        return response.json();
+      })
+      .then(() => {
+        window.location.href = `${window.location.origin}/interviews`;
+      })
+      .then((res) => {
```
Comment (estefrare) [inline src/Components/Interviews/Form/index.js]: remove this `.then`

File: src/Components/Interviews/Form/index.js [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+        console.log('res', res);
+      })
+      .catch((error) => {
+        console.log('err', error);
```
Comment (estefrare) [inline src/Components/Interviews/Form/index.js]: show errors in the UI

File: src/Components/Interviews/Form/index.js [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+      });
+  };
+  useEffect(() => {
+    fetch(`${process.env.REACT_APP_API}/postulants`)
+      .then((response) => response.json())
+      .then((res) => {
+        setPostulantsValue(res.data);
+      })
+      .catch((error) => {
+        console.log('err', error);
```
Comment (estefrare) [inline src/Components/Interviews/Form/index.js]: show errors in the UI, apply this on each `.catch()`

File: src/Components/Interviews/Form/index.js [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+          name="status"
+          type="text"
+          required
+          value={statusValue}
+          onChange={onChangeStatus}
+          className={styles.input}
+        >
+          <option value={''} disabled>
+            {'Select one'}
+          </option>
```
Comment (estefrare) [inline src/Components/Interviews/Form/index.js]:
```suggestion
          <option value={''} disabled>
            Select one
          </option>
```

**Review by estefrare — CHANGES_REQUESTED** [review]

**Review by estefrare — APPROVED** [review]

**Review by AgustinCarthery — APPROVED** [review]

**Pull request merged**